packages/checkout/sdk/src/smartCheckout/smartCheckout.ts

```diff
@@ -27,30 +24,46 @@ export const smartCheckout = async (
   provider: Web3Provider,
   itemRequirements: ItemRequirement[],
   transactionOrGasAmount: FulfillmentTransaction | GasAmount,
+  skipGasCalculation: boolean = false,
 ): Promise<SmartCheckoutResult> => {
   const ownerAddress = await provider.getSigner().getAddress();
 
   let aggregatedItems = itemAggregator(itemRequirements);
+  const erc20AllowancePromise = hasERC20Allowances(
+    provider,
+    ownerAddress,
+    aggregatedItems,
+  );
+  const erc721AllowancePromise = hasERC721Allowances(
+    provider,
+    ownerAddress,
+    aggregatedItems,
+  );
 
-  const erc20AllowancePromise = hasERC20Allowances(provider, ownerAddress, aggregatedItems);
-  const erc721AllowancePromise = hasERC721Allowances(provider, ownerAddress, aggregatedItems);
-
-  const resolvedAllowances = await measureAsyncExecution<{ sufficient: boolean, allowances: Allowance[] }[]>(
+  const resolvedAllowances = await measureAsyncExecution<
+  { sufficient: boolean; allowances: Allowance[] }[]
+  >(
     config,
     'Time to calculate token allowances',
     Promise.all([erc20AllowancePromise, erc721AllowancePromise]),
   );
 
-  const aggregatedAllowances = allowanceAggregator(resolvedAllowances[0], resolvedAllowances[1]);
-
-  const gasItem = await measureAsyncExecution<ItemRequirement | null>(
-    config,
-    'Time to run gas calculator',
-    gasCalculator(provider, aggregatedAllowances, transactionOrGasAmount),
+  const aggregatedAllowances = allowanceAggregator(
+    resolvedAllowances[0],
+    resolvedAllowances[1],
   );
-  if (gasItem !== null) {
-    aggregatedItems.push(gasItem);
-    aggregatedItems = itemAggregator(aggregatedItems);
+
+  let gasItem = null;
+  if (!skipGasCalculation) {
```

Comment:
I can see that the gasCalculator depends on `transactionOrGasAmount`, then making this optional will be enough as a way to flag gas is not required.

I like it simplifies the api, but also wonder if there is something I missing @mimi-imtbl @dreamoftrees ?

Reply:
Yep, I think in this context that makes sense and smartCheckout should skips the gasCalculator call.